res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1383 @@
+    // The Wheel control centers on 0x40; find out how much it's moved by.
+    return value - 0x40;
+};
+
+DDJSR2.wheelLedControl = function(channelOffset, color) {
+    const wheelLedBaseChannel = 0xBB;
+    midi.sendShortMsg(
+        wheelLedBaseChannel,
+        0x20+(channelOffset),
+        color
+    );
+};
+
+
+DDJSR2.pitchBendFromJog = function(group, movement) {
+    engine.setValue(group, "jog", movement / 5 * DDJSR2.jogwheelSensitivity);
+};
+
+DDJSR2.RingBufferView = class {
+    constructor(indexable, startIndex = 0) {
+        this.indexable = indexable;
+        this.index = startIndex;
+    }
+    advanceBy(n) {
+        this.index = script.posMod(this.index + n, this.indexable.length);
+        return this.current();
+    }
+    next() {
+        return this.advanceBy(1);
+    }
+    previous() {
+        return this.advanceBy(-1);
+    }
+    current() {
+        return this.indexable[this.index];
+    }
+};
+
+DDJSR2.BrowserContainer = function() {
+    this.browseKnob  = new components.ComponentContainer({
+        longPressTimer: 0,
+        longPressTimeout: 250,
+        trackColorCycleEnabled: false,
+        trackColorCycleHappened: false,
+        previewSeekEnabled: false,
+        previewSeekHappened: false,
+        turn: new components.Encoder({
+            group: "[Library]",
+            inKey: "MoveVertical",
+            inValueScale: DDJSR2.getRotaryDelta
+        }),
+        press: new components.Button({
+            group: "[Library]",
+            key: "GoToItem"
+        }),
+        shiftPress: new components.Button({
+            group: "[Library]",
+            input: function(channel, control, value, _status, _group) {
+                if (!value) { // On release, not press. Allows for track color cycling.
+                    this.trackColorCycleEnabled = false;
+                    if (!this.trackColorCycleHappened) {
+                        if (!engine.getValue("[PreviewDeck1]", "play")) {
+                            engine.setValue("[PreviewDeck1]", "LoadSelectedTrackAndPlay", 1);
+                        } else {
+                            script.triggerControl("[PreviewDeck1]", "stop");
+                        }
+                    } else {
+                        this.trackColorCycleHappened = false;
+                    }
+                } else {
+                    this.trackColorCycleEnabled = true;
+                }
+            }.bind(this)
+        }),
+        shiftTurn: new components.Encoder({
+            group: "[PreviewDeck1]",
+            input: function(channel, control, value, _status, _group) {
+                const rotateValue = DDJSR2.getRotaryDelta(value);
+                if (!this.trackColorCycleEnabled) {
+                    const oldPos = engine.getValue("[PreviewDeck1]", "playposition");
+                    const newPos = Math.max(0, oldPos + (0.05 * rotateValue));
+                    engine.setValue("[PreviewDeck1]", "playposition", newPos);
```

Comment:
alternatively, depending on the encoder direction, just beatjumping 4-8 beats in either direction would also work nicely, wdyt?